packages/block-editor/src/components/block-list/use-block-props/use-focus-handler.js

```diff
@@ -42,7 +42,7 @@ export function useFocusHandler( clientId ) {
 
 				// If an inner block is focussed, that block is resposible for
 				// setting the selected block.
-				if ( ! isInsideRootBlock( node, event.target ) ) {
```

Comment:
I actually misread this. It's weird that this is not true for the demo content image.